.github/workflows/publish.yaml

```diff
@@ -0,0 +1,29 @@
+name: Publish
+
+on:
+  push:
+    branches: [master]
+    tags: ['*']
+
+jobs:
+  publish-api-doc:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone the module
+        uses: actions/checkout@v2
+
+      - name: Setup Doxygen
+        run: sudo apt install -y doxygen
+
+      - name: Build module API documentation using Doxygen
+        run: |
+            cmake .
+            make module_api
+            sed -e "s%@PROJECT_.\+_DIR@/%%" Doxyfile.API.in > Doxyfile.API
+            doxygen Doxyfile.API
```

Comment:
I'm just curious: how about a custom css theme to make the result a bit more nice?

I searched the web across and found those ones:

https://jothepro.github.io/doxygen-awesome-css/ (just css file)
https://mcss.mosra.cz/documentation/doxygen/ (a script to process doxygen generated files)

Feel free to ignore/postpone it.